src/classes/OPP_SendAcknowledgmentBTN_CTRL.cls

```diff
@@ -51,60 +56,75 @@ public with sharing class OPP_SendAcknowledgmentBTN_CTRL {
     public Boolean redirect { get; set; }
 
     /*******************************************************************************************************
-    * @description the list of Opportunities to be acknowledged
+    * @description List of Opportunities that should receive email acknowledgements 
     */
-    private list<Opportunity> listOpp { 
+    @TestVisible
+    private List<Opportunity> oppsToAcknowledge { 
         get {
-            if (listOpp == null)
-                listOpp = new list<Opportunity>(); 
-            return listOpp;
+            if (oppsToAcknowledge == null) {
+                oppsToAcknowledge = [
+                    SELECT Acknowledgment_Status__c 
+                    FROM Opportunity
+                    WHERE Id IN :oppIds
+                ]; 
+            }
+            return oppsToAcknowledge;
         }
         set; 
     }
 
+    /*******************************************************************************************************
+    * @description Ids of Opportunities that should receive email acknowledgements
+    */
+    private Set<Id> oppIds = new Set<Id>();
+
     /*******************************************************************************************************
     * @description the list of Opportunities that were not acknowledged
     */
-    public list<Opportunity> listOppNotSent { 
-        get {
-            if (listOppNotSent == null)
-                listOppNotSent = new list<Opportunity>(); 
-            return listOppNotSent;
-        }
-        set; 
-    }
+    public List<Opportunity> listOppNotSent { get; set; }
 
     /*******************************************************************************************************
     * @description the count of Opportunities that were acknowledged
     */
-    public integer cSent { get; set; }
+    public Integer cSent { get; set; }
+
+    /*******************************************************************************************************
+    * @description the count of Opportunities where acknowledgements were already sent
+    */
+    public Integer cAlreadySent { get; set; }
```

Comment:
I realize that the other vars use the "c" prefix, though this is not common practice for our controllers. I'd guess it stands for "count". I think it would be better to clean up the public vars to remove the "c" prefix, so these become something like `sentCount`, `alreadySentCount` and `notSentCount`. I'm open to another naming pattern as well.